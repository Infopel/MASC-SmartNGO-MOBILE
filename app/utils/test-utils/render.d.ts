export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
