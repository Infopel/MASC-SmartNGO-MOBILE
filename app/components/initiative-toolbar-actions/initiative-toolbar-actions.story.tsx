import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { InitiativeToolbarActions } from "./initiative-toolbar-actions"

storiesOf("InitiativeToolbarActions", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <InitiativeToolbarActions m='2' />
      </UseCase>
    </Story>
  ))
