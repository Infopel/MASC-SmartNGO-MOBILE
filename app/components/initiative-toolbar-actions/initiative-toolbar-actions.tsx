import { Icon } from "components/icon/icon"
import { translate } from "i18n/translate"
import { Box, IBoxProps, IconButton, Menu } from "native-base"
import { navigate } from "navigators/navigation-utilities"
import * as React from "react"

export interface InitiativeToolbarActionsProps extends IBoxProps {
  initiativeId: string
}

/**
 * Describe your component here
 */
export function InitiativeToolbarActions({ initiativeId, ...rest }: InitiativeToolbarActionsProps) {


  return (
    <Box alignItems="center">
      <Menu trigger={triggerProps => {
        return <IconButton
          p={1} {...triggerProps} testID="back" colorScheme={'text'} icon={<Icon size={'xl'} name='dots-vertical' type='material-community' />}
        />
      }}>
        <Menu.Item onPress={()=> navigate('action-plan', {initiativeId})}>{translate('context-menu.view-action-plan')}</Menu.Item>
        <Menu.Item onPress={()=> navigate('monitoring-visits', {initiativeId})}>{translate('context-menu.view-monitoring-visit')}</Menu.Item>
      </Menu>
    </Box>
  )
}
