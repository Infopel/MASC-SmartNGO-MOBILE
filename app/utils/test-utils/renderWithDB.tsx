jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  