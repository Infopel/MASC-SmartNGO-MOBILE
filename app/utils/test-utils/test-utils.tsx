import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

