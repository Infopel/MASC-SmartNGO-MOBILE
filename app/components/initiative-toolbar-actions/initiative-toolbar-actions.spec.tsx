import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { InitiativeToolbarActions } from "./initiative-toolbar-actions"



describe("<InitiativeToolbarActions>",()=>{
  it("render", async function () {
    const component = <InitiativeToolbarActions></InitiativeToolbarActions>
  
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
