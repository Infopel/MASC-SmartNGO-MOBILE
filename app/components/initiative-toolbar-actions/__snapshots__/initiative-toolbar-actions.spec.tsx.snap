// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InitiativeToolbarActions> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    aria-haspopup="menu"
    collapsable={false}
    colorScheme="text"
    dataSet={Object {}}
    focusable={true}
    nativeID="1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDownCapture={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 4,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 4,
      }
    }
    testID="back"
  >
    <Text />
  </View>
</View>
`;
